common/web/keyboard-processor/src/keyboards/keyboard.ts

```diff
@@ -30,15 +30,42 @@ export interface VariableStoreDictionary {
   [name: string]: string;
 };
 
+export type KeyboardObject = {
```

Comment:
Is this the only definition we have anywhere for this type? It feels like it belongs in common/web/types in the future, with copious comments...

Reply:
So far as I can tell, yes.  I always _did_ want to write up a formal type for it, and this push gave the perfect excuse.

I'd be happy for the type to be hoisted to `common/web/types`; be my guest!

Reply:
I just went ahead and added documentation for all fields.